tedana/metrics/external.py

```diff
@@ -0,0 +1,650 @@
+        match the number of timepoints in the fMRI time series
+    external_regressor_config : :obj:`dict`
+        Information describing the external regressors and
+        method to use for fitting and statistical tests
+    mixing : (T x C) array_like
+        Mixing matrix for converting input data to component space, where `C`
+        is components and `T` is the same as in `data_cat`
+
+    Returns
+    -------
+    comptable : (C x X) :obj:`pandas.DataFrame`
+        Component metric table. Same as inputted, with additional columns
+        for metrics related to fitting the external regressors
+    """
+    n_time = mixing.shape[0]
+
+    # If the order of polynomial detrending is specified, then pass to make_detrend_regressors
+    # otherwise the function sets a detrending polynomial order
+    if external_regressor_config["detrend"] is True:
+        detrend_regressors = make_detrend_regressors(n_time, polort=None)
+    elif (
+        isinstance(external_regressor_config["detrend"], int)
+        and external_regressor_config["detrend"] > 0
+    ):
+        detrend_regressors = make_detrend_regressors(
+            n_time, polort=external_regressor_config["detrend"]
+        )
+    else:
+        LGR.warning(
+            "External regressor fitted without detrending fMRI time series. Only removing mean"
+        )
+        detrend_regressors = make_detrend_regressors(n_time, polort=0)
+
+    if external_regressor_config["calc_stats"].lower() == "f":
+        # external_regressors = pd.concat([external_regressors, detrend_regressors])
+        comptable = fit_mixing_to_regressors(
+            comptable, external_regressors, external_regressor_config, mixing, detrend_regressors
+        )
+    else:
+        # This should already be valided by this point, but keeping the catch clause here
+        # since this would otherwise just return comptable with no changes, which would
+        # make a hard-to-track error
+        raise ValueError(
+            "calc_stats for external regressors in decision tree is "
+            f"{external_regressor_config['calc_stats'].lower()}, which is not valid."
+        )
+
+    return comptable
+
+
+def make_detrend_regressors(n_time, polort=None):
+    """
+    Create polynomial detrending regressors to use for removing slow drifts from data.
+
+    Parameters
+    ----------
+    n_time : :obj:`int`
+        The number of time point in the fMRI time series
+    polort : :obj:`int` or :obj:`NoneType`
+        The number of polynomial regressors to create (i.e. 3 is x^0, x^1, x^2)
+        If None, then this is set to 1+floor(n_time/150)
+
+    Returns
+    -------
+    detrend_regressors: (n_time x polort) :obj:`pandas.DataFrame`
+        Dataframe containing the detrending regressor time series
+        x^0 = 1. All other regressors are zscored so that they have
+        a mean of 0 and a stdev of 1.
+        Dataframe column labels are polort0 - polort{polort-1}
```

Comment:
This is where it happens in gscontrol_raw

https://github.com/ME-ICA/tedana/blob/325011e3c65cc6f0ea43d1c4d9b8e0367297e091/tedana/gscontrol.py#L66-L69

It's using Legendre polynomials while I'm just using a + bx + cx² + dx³ + ex⁴. Changing would involve agreeing on one or the other and moving code out of here and `gscontrol_raw` I'd prefer just keeping this in two places, but could be convinced otherwise.